packages/teams-js/test/public/meeting.spec.ts

```diff
@@ -1372,6 +1372,53 @@ describe('meeting', () => {
       });
     });
 
+    describe('joinMeeting', () => {
```

Comment:
You are missing some unit tests for your code, specifically it doesn't look like you are testing that the right parameters are being sent up to the host sdk in the message and it doesn't look like you are having the unit tests respond to the message (which would resolve the Promise), but are instead letting the unit test timeout and then resolve the promise that way (which isn't actually testing your code correctly).

[This file](https://github.com/OfficeDev/microsoft-teams-library-js/blob/main/packages/teams-js/test/public/search.spec.ts) is a good example of how unit tests should be put together. `search.CloseSearch` is similar to the function you are adding in that the return type is `Promise<void>`.

You should have unit tests for your function in both the 'framed' and 'frameless` scenarios. If you look in `meeting.spec.ts` you can see that there are sections of unit tests for each of these already (look for `describe('framed', ()` and `describe('frameless', ()`

For both framed and frameless tests, you can see the message being passed out of TeamsJS and up to the host by writing code in your test like this:

```typescript
const joinMeetingPromise = meeting.joinMeeting(args);
const joinMeetingMessage = utils.findMessageByFunc('meeting.joinMeeting');
expect(joinMeetingMessage).not.toBeNull();
```

`joinMeetingMessage.args` will have whatever parameters are passed up to the host sdk in it, so you can verify that the right values are there (for example, source should be Other when no source is passed into the function).

For *framed* tests, you can respond to the message using code like this:

```typescript
utils.respondToMessage(joinMeetingMessage, args); // args is whatever arguments the host sdk should pass back to respond to the promise and get it to resolve
await joinMeetingPromise;
```

This will make sure that the promise resolves correctly in the framed state. For frameless tests, instead do this:

```typescript
utils.respondToFramelessMessage({
  data: {
    id: joinMeetingMessage?.id,
    args: [], // whatever args are passed down from the host sdk go here
  },
} as DOMMessageEvent);
await joinMeetingPromise;
```

Let me know if that gives you any trouble